src/fixedSizeSquareMatrixOpsImpl.hpp

```diff
@@ -1162,6 +1212,56 @@ struct SquareMatrixOps< 3 >
     dstMatrix[ 2 ][ 1 ] = srcSymMatrix[ 3 ];
   }
 
+  /**
+   * @brief Determine the polar decomposition of the 3x3 matrix @p matrix
+   * @tparam DST_MATRIX The type of @p R.
+   * @tparam MATRIX The type of @p matrix.
+   * @param R The resultant rotation matrix.
+   * @param matrix The matrix to be decomposed.
+   * @details The polar decomposition returns a rotation matrix such that @p R . U = V . @p R = @p matrix.
+   *   This is done using Higham's iterative algorithm.
+   */
+  template< typename DST_MATRIX, typename MATRIX >
```

Comment:
I'm probably just not seeing it, but is there a difference between the algorithm for 2x2 and 3x3 matrices? If not you should probably just write a single function. If it works for any size square matrix then it should be made general and moved out of `fixedSizeSquareMatrixOps`

Reply:
The algo does work for any nonsingular square matrix, but requires an inverse calculation which I believe is only explicitly implemented for 2x2 and 3x3 matrices.

Reply:
Yeah my bad, I missed that. I would still suggest that you write one function to work on both 2x2 and 3x3 matrices.

Reply:
ok, will do

Reply:
So it turns out I'm not sure how to implement this. I moved `polarDecomposition` to the unspecialized version of `SquareMatrixOps`, but then the compiler complains about `polarDecomposition` not being implemented in the specialized versions. It seems this problem is typically overcome using inheritance, but that seems like overkill here. Is there an easier way to do this?

Reply:
Yeah, jut make a free function and then have the specialized versions just be a thin wrapper.